projects/tests/test_section.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+from django.contrib.auth import get_user_model
+from rest_framework import serializers
+
+from projects.models import Attribute, AttributeValueChoice
+from projects.serializers.section import (
+    create_section_serializer,
+    _is_attribute_required,
+    _get_serializer_field_data,
+)
+
+
+@pytest.mark.django_db()
+def test_create_section_serializer(
+    f_project_section_1, f_project_section_attribute_1, f_project_section_attribute_2
+):
+    serializer = create_section_serializer(f_project_section_1)
+    fields = serializer._declared_fields
+    assert len(fields) == 2
+
+
+@pytest.mark.django_db()
+def test_is_attribute_required(f_project_section_attribute_1):
+    required = _is_attribute_required(f_project_section_attribute_1)
+    assert required is False
+
+    # required = True
+    f_project_section_attribute_1.required = True
+    required = _is_attribute_required(f_project_section_attribute_1)
+    assert required is True
+
+    # required = True, generated = True
+    f_project_section_attribute_1.generated = True
+    required = _is_attribute_required(f_project_section_attribute_1)
+    assert required is False
+
+    # required = True, generated = False, Boolean field
+    f_project_section_attribute_1.generated = False
+    f_project_section_attribute_1.attribute.value_type = Attribute.TYPE_BOOLEAN
+    assert required is False
```

Comment:
This is probably testing checking the value set in the previous section i.e. is this part missing `required = _is_attribute_required(f_project_section_attribute_1)` ?

Reply:
Good catch! Thanks